### ISSUE TITLE
[Node feature via Dockerfile] Tasks missing NVM from `$PATH`

### ISSUE THREAD

**Issue by aaronadamsCA** [microsoft/vscode-remote-release#6599]
- VSCode Version: 1.66.2
- Local OS Version: Windows 11 Pro 21H2
- Local chip architecture: x64
- Reproduces in: Both Codespaces and Remote - Containers

Steps to Reproduce:

1. **Clone Repository in Container Volume** > https://github.com/aaronadamsCA/vscode-devcontainer-features-issue
2. Wait until Node.js is downloaded and installed.
3. Open the **NPM Scripts** pane and run the `null` script.

Expected: The script exits successfully.
Actual: Error message "Git: Yarn requires Node.js 4.0 or higher to be installed."

It seems the VS Code Git process is missing NVM from `$PATH`.

Additional information:

- At the CLI, `$PATH` is correct and `git commit` works fine.
- After **Reload Window**, the script runs fine.
  - This workaround is not applicable to Codespaces, since there is no **Reload Window** command.
- Branch [node-as-feature](https://github.com/aaronadamsCA/vscode-devcontainer-features-issue/tree/node-as-feature) demonstrates that when Node.js support is added as a `devcontainer.json` feature instead of a `Dockerfile` script, everything works fine; but I'd prefer to use the script approach, since it's easier to ship a built image.

**Comment by aaronadamsCA** [comment]
I couldn't figure out how to do that without an error; but I did figure out that if I add NVM as a `devcontainer.json` feature instead of a `Dockerfile` script, it works fine on the first build.

I would prefer a Dockerfile-based approach; is it possible to fix the script to work like the feature?

**Comment by Chuxel** [comment]
The feature uses the script under the hood, but it sets the env vars mentioned in the README for the script.

https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/node.md

It looks like you are missing these:

```
ENV NVM_DIR="/usr/local/share/nvm"
ENV NVM_SYMLINK_CURRENT=true \
    PATH=${NVM_DIR}/current/bin:${PATH}
```

If this is not set, then node will only be available when nvm is sourced.... which happens for interactive shells but not non-interactive ones (and there isn't really a way to do it unfortunatley).

**Comment by aaronadamsCA** [comment]
I updated the original issue report to keep things simple; the problem, specifically, is the value of `$PATH` on the first run when adding Node.js as a script, rather than as a feature.

**Comment by aaronadamsCA** [comment]
@Chuxel I had just figured that out too, while you were posting. Thank you for confirming it was where I went wrong!

I think I missed it because I read https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/README.md#downloading-the-script-with-curl--wget-instead and erroneously thought I could run the `.sh` for any script and be done with it. I failed to RTFM as directed on that page and check if the script itself had any prerequisites.

Appreciate the assistance!
